enclave/core/sgx/calls.c

```diff
@@ -161,7 +316,16 @@ static oe_result_t _handle_init_enclave(uint64_t arg_in)
 
         if (_once == false)
         {
-            oe_enclave_t* enclave = (oe_enclave_t*)arg_in;
+            oe_enclave_t* enclave;
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+            oe_enclave_with_initialization_data_t* enclave_plus =
+                (oe_enclave_with_initialization_data_t*)arg_in;
+            enclave = enclave_plus->enclave;
+
+            OE_CHECK(_check_and_load_initialization_data(enclave_plus));
```

Comment:
Would be safer to call this function after basic initalization functions are done, for example, after line 348.

Reply:
Why would that be safer?

Reply:
`_check_and_load_initilization_data()` use malloc and other functions which need to be initialized. If OE enclave initialization implementation changes and moves things around, invoking malloc or other functions before a set of initialization functions might fail.

Reply:
The memory allocator is why we used to do part of this in `init.c`, but that's not necessary anymore. If things get moved around, this may also have to move, but I prefer to do this as early as possible to avoid any unexpected effects.